zerver/tests/test_decorators.py

```diff
@@ -1604,6 +1604,14 @@ def test_internal_requests_with_broken_secret(self) -> None:
                 self.internal_notify(True, request)
             self.assertEqual(access_denied_error.exception.http_status_code, 403)
 
+        with self.settings(SHARED_SECRET=None):
+            with self.assertRaises(AccessDeniedError), self.assertLogs(level="ERROR") as m:
+                self.assertFalse(authenticate_notify(request))
+                self.internal_notify(True, request)
```

Comment:
I think it's specifically this line that's supposed to raise `AccessDeniedError`, right? Then I think we should have `assertRaises` just  around this specific line